projects/reservoir/fv3/save_ranks.py

```diff
@@ -0,0 +1,133 @@
+import argparse
+import intake
+import logging
+import os
+from tempfile import NamedTemporaryFile
+import toolz
+
+from fv3fit.reservoir import CubedsphereDivider
+import vcm
+
+logging.basicConfig()
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def _get_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "data_path", type=str, default=None, help=("Path with training data zarr."),
+    )
+    parser.add_argument(
+        "output_path",
+        type=str,
+        help="Local or remote path where output will be written.",
+    )
+    parser.add_argument(
+        "layout_width",
+        type=int,
+        default=1,
+        help=(
+            "Width of rank layout within each tile, ex. layout_width=2 corresponds "
+            "to (2,2) layout with four ranks per tile. Defaults to 1 rank per tile."
+        ),
+    )
+    parser.add_argument(
+        "overlap",
+        type=int,
+        help="Number of grid cells overlapping between training subdomains.",
+    )
+    parser.add_argument(
+        "--start-time",
+        type=str,
+        default=None,
+        help=(
+            "First timestep in time series. Provide a string 'YYYYMMDD.HHMMSS'. "
+            "If not provided, will use the first timestep in dataset."
+        ),
+    )
+    parser.add_argument(
+        "--stop-time",
+        type=str,
+        default=None,
+        help=(
+            "Last timestep in time series. Provide a string 'YYYYMMDD.HHMMSS'. "
+            "If not provided, will use the last timestep in dataset."
+        ),
+    )
+    parser.add_argument(
+        "--time-chunks",
+        type=int,
+        default=None,
+        help=("Number of timesteps to save per rank netcdf."),
+    )
+    parser.add_argument("--variables", type=str, nargs="+", default=[])
+
+    return parser
+
+
+def get_ordered_dims_extent(dims: dict):
+    # tile must be first in dims list
+    dims_, extent_ = [], []
+
+    for d in ["tile", "x", "y"]:
+        dims_.append(d)
+        extent_.append(dims.pop(d))
+    for k, v in dims.items():
+        dims_.append(k)
+        extent_.append(v)
+    return dims_, extent_
+
+
+if __name__ == "__main__":
```

Comment:
Non-blocking suggestion: You might pass on this for this time but I'm spending a fair amount of review time manually "testing" that the argument parsing into variables done in this main block is correct. It would be better if you could isolate it into a unit test that takes in the command-line arguments and tested that the command-line args have the expected effect on internal variables like `tstart`, `tstop`, and `time_chunks`.